cosmos/miner/miner.go

```diff
@@ -68,11 +85,34 @@ func (m *Miner) Init(serializer EnvelopeSerializer) {
 func (m *Miner) PrepareProposal(
```

Comment:
@alexanderbez is this safe?

Reply:
I haven't dug into the specifics of each individual method call, but since `PrepareProposal` state is discarded upon round timeout, any state modifications should be safe assuming you use the provided `ctx` (which it looks like you are!).

Reply:
If you'd like, I can dive into the individual method calls for further analysis.